akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1397,10 +1397,14 @@ private[stream] object Collect {
 
       private var inFlight = 0
       private var buffer: BufferImpl[Out] = _
+      private var invokeFutureCB: Try[Out] => Unit = _
```

Comment:
Couldn't we even initialize it right away and make it a val?

Reply:
As the document said , the `getAsyncCallback` is better to be called from `preStart`，there are some race condition in  the constructor，is that true?

Reply:
Sharing the AsyncCallback from the constructor risks race conditions, therefore it is recommended to use the preStart() lifecycle hook instead.

Sharing the AsyncCallback from the constructor risks race conditions, therefore it is recommended to use the preStart() lifecycle hook instead.

Reply:
No sharing of the callback (as in with an outside object at construction) going on here, only calling it triggered by stream push, so it is safe

Reply:
good to know that.